src/python/pants/init/specs_calculator.py

```diff
@@ -45,26 +37,25 @@ def parse_specs(
     """Parse raw string specs into a Specs object."""
     build_root = build_root or get_buildroot()
     spec_parser = CmdLineSpecParser(build_root)
-    specs = OrderedSet(spec_parser.parse_spec(spec_str) for spec_str in raw_specs)
-    address_specs = AddressSpecs(
-      dependencies=(spec for spec in specs if isinstance(spec, AddressSpec)),
+
+    address_specs: OrderedSet = OrderedSet()
+    filesystem_specs: OrderedSet = OrderedSet()
+    for spec_str in raw_specs:
+      parsed_spec = spec_parser.parse_spec(spec_str)
+      if isinstance(parsed_spec, AddressSpec):
+        address_specs.add(parsed_spec)
+      else:
+        filesystem_specs.add(parsed_spec)
```

Comment:
I changed this to a more imperative style for performance. Before, the two generator expressions meant that we would iterate through `specs` twice. Now, we only iterate through it once.